src/ore/src/tracing.rs

```diff
@@ -79,6 +79,40 @@ pub struct StderrLogConfig {
     pub filter: Targets,
 }
 
+/// A callback that can be used to modify stderr logging filters
+pub struct StderrFilterCallback {
+    callback: Arc<dyn Fn(Targets) -> Result<(), anyhow::Error> + Send + Sync>,
+}
+
+impl StderrFilterCallback {
+    /// Updates stderr log filtering with `targets`
+    pub fn call(&self, targets: Targets) -> Result<(), anyhow::Error> {
+        (self.callback)(targets)
```

Comment:
dont we also need something like: https://github.com/MaterializeInc/materialize/pull/13462 to make sure its passed through the tracing orchestrator?

Reply:
hm... I'm not sure. since the CLI argument for `environmentd` is static, it seems like a bit of a smell that it could produce downstream resources like `storaged`/`computed` with a different log filter than itself. this would mean that when `environmentd` restarts, its log filter will be reset, and it'll recompute different Kube definitions for its downstream resources and potentially restart them

the current structure of using an HTTP call to dynamically updating tracing/log filters seems more suited for targeted changes to a running pod, with no expectations of permanence. separately, and I think addressing what you're getting at, it seems like it'd be useful to have a way to declaratively set a log filter override for a whole environment, so that all envd/computed/storaged pick up the changes together. I could imagine this value being stored in a ConfigMap somewhere that the pods either mount or poll with some frequency

Reply:
@pH14 hmm thats true, I didn't think about envd restarting when i made this logic for the otel stuf...that said, i feel like...having to send a curl request for every storaged/computed you start after you change the filter for envd is annoying...but perhaps `bin/mz-admin enable-otel`/change filter or whatever we call it is easy enough that it doesnt matter. The place this gets annoying is when you want to debug the startup or storaged or whatever...

lets leave it for now, (with a comment explaining the choice!), and think about what we want (`ConfigMap` seems nice)

Reply:
yeah the startup/persisted override issue is where a `ConfigMap` would really shine. on startup we'd read the CLI value, override with the `ConfigMap` (if any), and then watch it for any changes. and if it's shared for an env, then we don't need to muck with any CLI args going into storaged/computed and wait for a restart to take effect

Reply:
I find @pH14's argument very compelling!